pkg/microservices/microservices.go

```diff
@@ -0,0 +1,124 @@
+package microservices
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/hobbyfarm/gargantua/pkg/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type MicroService struct {
+	name string
+}
+
+var (
+	AuthN      = MicroService{"authn-service"}
+	AuthR      = MicroService{"authr-service"}
+	User       = MicroService{"user-service"}
+	Rbac       = MicroService{"rbac-service"}
+	AccessCode = MicroService{"accesscode-service"}
+	Setting    = MicroService{"setting-service"}
+	Unknown    = MicroService{""}
+)
+
+func (m MicroService) getUrl() (string, error) {
+	// Create a Kubernetes clientset
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", fmt.Errorf("error retrieving InClusterConfig")
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", fmt.Errorf("error creating kubernetes clientset")
+	}
+
+	ctx := context.Background()
+
+	// Get the endpoints object for the service
+	endpoints, err := clientset.CoreV1().Endpoints(util.GetReleaseNamespace()).Get(ctx, m.name, metav1.GetOptions{})
```

Comment:
Can you tell me a little more about your thought process here? Usually getting the list of endpoints isn't recommended for this type of activity. We want to just call the service on a specific port - which allows us to hide the implementation details. Honestly decoupling it from Kubernetes if we can in case we want to implement this service on a different hosting provider/methodology.

Reply:
We could either use the same grpc port for every service or use a DNS SRV Request to discover the gprc port of each service.

Reply:
Eamon and i decided we are using the port `8080` for grpc endpoints from now on